src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -0,0 +1,286 @@
+    internal static class NegotiateStreamPal
+    {
+        // value should match the Windows sspicli NTE_FAIL value
+        // defined in winerror.h
+        private const int NTE_FAIL = unchecked((int)0x80090020);
+
+        internal static IIdentity GetIdentity(NTAuthentication context)
+        {
+            Debug.Assert(!context.IsServer, "GetIdentity: Server is not supported");
+
+            string name = context.Spn;
+            string protocol = context.ProtocolName;
+
+            return new GenericIdentity(name, protocol);
+
+        }
+
+        internal static string QueryContextAssociatedName(SafeDeleteContext securityContext)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static string QueryContextAuthenticationPackage(SafeDeleteContext securityContext)
+        {
+            return NegotiationInfoClass.Kerberos;
+        }
+
+        internal static int QueryMaxTokenSize(string package)
+        {
+            // This value is not used on Unix
+            return 0;
+        }
+
+        internal static string QueryContextClientSpecifiedSpn(SafeDeleteContext securityContext)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static SafeFreeCredentials AcquireDefaultCredential(string package, bool isServer)
+        {
+            return AcquireCredentialsHandle(package, isServer, new NetworkCredential(string.Empty, string.Empty, string.Empty));
+        }
+
+        internal static SafeFreeCredentials AcquireCredentialsHandle(string package, bool isServer, NetworkCredential credential)
+        {
+            if (isServer)
+            {
+                throw new PlatformNotSupportedException(SR.net_nego_server_not_supported);
+            }
+
+            bool isNtlm = string.Equals(package, NegotiationInfoClass.NTLM);
+
+            if (isNtlm)
+            {
+                throw new PlatformNotSupportedException(SR.net_nego_ntlm_not_supported);
+            }
+
+            // Note: In client case, equivalent of default credentials is to use previous, cached Kerberos TGT to get service-specific ticket.
+            return (string.IsNullOrEmpty(credential.UserName) || string.IsNullOrEmpty(credential.Password)) ?
+                new SafeFreeNegoCredentials(string.Empty, string.Empty, string.Empty) :
+                new SafeFreeNegoCredentials(credential.UserName, credential.Password, credential.Domain);
+        }
+
+        internal static SecurityStatusPal InitializeSecurityContext(
+            SafeFreeCredentials credentialsHandle,
+            ref SafeDeleteContext securityContext,
+            string spn,
+            ContextFlagsPal requestedContextFlags,
+            SecurityBuffer[] inSecurityBufferArray,
+            SecurityBuffer outSecurityBuffer,
+            ref ContextFlagsPal contextFlags)
+        {
+            // TODO (Issue #3718): The second buffer can contain a channel binding which is not supported
```

Comment:
The comment says that the second buffer could be a channel binding.  What checked the first buffer to not be a channel binding?

`Debug.Assert(inSecurityBufferArray == null || inSecurityBufferArray[0].type == SecurityBufferType.Data);`

Unless it's Stream, or something else, or can be multiple things, ...